smt/expr.cpp

```diff
@@ -731,6 +731,77 @@ expr expr::fdiv(const expr &rhs) const {
   return Z3_mk_fpa_div(ctx(), rm, ast(), rhs());
 }
 
+expr expr::foeq(const expr &rhs) const {
+  C(rhs);
+  return ford(rhs) && Z3_mk_fpa_eq(ctx(), ast(), rhs());
+}
+
+expr expr::fogt(const expr &rhs) const {
+  C(rhs);
+  return ford(rhs) && Z3_mk_fpa_eq(ctx(), ast(), rhs());
+}
+
+expr expr::foge(const expr &rhs) const {
+  C(rhs);
+  return ford(rhs) && Z3_mk_fpa_geq(ctx(), ast(), rhs());
+}
+
+expr expr::folt(const expr &rhs) const {
+  C(rhs);
+  return ford(rhs) && Z3_mk_fpa_lt(ctx(), ast(), rhs());
+}
+
+expr expr::fole(const expr &rhs) const {
+  C(rhs);
+  return ford(rhs) && Z3_mk_fpa_leq(ctx(), ast(), rhs());
+}
+
+expr expr::fone(const expr &rhs) const {
+  C(rhs);
+  return ford(rhs) && (Z3_mk_not(ctx(), Z3_mk_fpa_eq(ctx(), ast(), rhs())));
```

Comment:
this is not safe because doesn't increment the reference counter properly. To negate you need to do:
`... && !expr(Z3_mk_fpa_...)`

(the explicit cast to expr is needed unfortunately)